share/smack/prelude.py

```diff
@@ -0,0 +1,129 @@
+INLINE_BINARY_COMP = lambda bitwidth, name, body: Function('inline', None, '.'.join([name, mk_int(bitwidth), 'bool']), [mk_int(bitwidth), mk_int(bitwidth)], 'bool', body)
+INLINE_BINARY_BV_COMP = lambda bitwidth, name, body: Function('inline', None, '.'.join([name, mk_bv(bitwidth), 'bool']), [mk_bv(bitwidth), mk_bv(bitwidth)], 'bool', body)
+BVBUILTIN_BINARY_COMP = lambda bitwidth, builtin_name, name: Function('bvbuiltin', builtin_name, '.'.join([name, mk_bv(bitwidth), 'bool']), [mk_bv(bitwidth), mk_bv(bitwidth)], 'bool', None)
+INLINE_BINARY_PRED = lambda bitwidth, name: Function('inline', None, '.'.join([name, mk_int(bitwidth)]), [mk_int(bitwidth), mk_int(bitwidth)], 'i1', 'if ' + '.'.join([name, mk_int(bitwidth), 'bool']) + '(i1, i2) then 1 else 0')
+INLINE_BINARY_BV_PRED = lambda bitwidth, name: Function('inline', None, '.'.join([name, mk_bv(bitwidth)]), [mk_bv(bitwidth), mk_bv(bitwidth)], 'bv1', 'if ' + '.'.join([name, mk_bv(bitwidth), 'bool']) + '(i1, i2) then 1bv1 else 0bv1')
+SAFE_LOAD_OP = lambda bitwidth, body: Function('inline', None, '.'.join(['$load', mk_int(bitwidth)]), ['[ref] ' + mk_int(bitwidth), 'ref'], mk_int(bitwidth), body)
+SAFE_STORE_OP = lambda bitwidth, body: Function('inline', None, '.'.join(['$store', mk_int(bitwidth)]), ['[ref] ' + mk_int(bitwidth), 'ref', mk_int(bitwidth)], '[ref] ' + mk_int(bitwidth), body)
+SAFE_LOAD_BV_OP = lambda bitwidth, body: Function('inline', None, '.'.join(['$load', mk_bv(bitwidth)]), ['[ref] ' + mk_bv(bitwidth), 'ref'], mk_bv(bitwidth), body)
+SAFE_STORE_BV_OP = lambda bitwidth, body: Function('inline', None, '.'.join(['$store', mk_bv(bitwidth)]), ['[ref] ' + mk_bv(bitwidth), 'ref', mk_bv(bitwidth)], '[ref] ' + mk_bv(bitwidth), body)
+INLINE_BVBUILTIN_BINARY_SELECT = lambda bitwidth, name, name2: Function('inline', None, '.'.join([name, mk_bv(bitwidth)]), [mk_bv(bitwidth), mk_bv(bitwidth)], mk_bv(bitwidth), 'if ' + '.'.join([name2, mk_bv(bitwidth), 'bool']) + '(i1, i2) then i1 else i2')
+INLINE_BVBUILTIN_BINARY_PRED = lambda bitwidth, name: Function('inline', None, '.'.join([name, mk_bv(bitwidth)]), [mk_bv(bitwidth), mk_bv(bitwidth)], 'bv1', 'if ' + '.'.join([name, mk_bv(bitwidth), 'bool']) + '(i1,i2) then 1bv1 else 0bv1')
+INLINE_CONVERSION = lambda bitwidth2, bitwidth1, name, body: Function('inline', None, '.'.join([name, mk_int(bitwidth1), mk_int(bitwidth2)]), [mk_int(bitwidth1)], mk_int(bitwidth2), body)
+TRUNC_OP = lambda bitwidth1, bitwidth2: Function('inline', None, '.'.join(['$trunc', mk_int(bitwidth1), mk_int(bitwidth2)]), [mk_int(bitwidth1)], mk_int(bitwidth2), 'i1')
+TRUNC_BV_OP = lambda bitwidth1, bitwidth2: Function('inline', None, '.'.join(['$trunc', mk_bv(bitwidth1), mk_bv(bitwidth2)]), [mk_bv(bitwidth1)], mk_bv(bitwidth2), 'i1[' + str(bitwidth2) + ':0]')
+
+sizes = [1, 8, 16, 24, 32, 40, 48, 56, 64, 88, 96, 128]
+binary_ops = ['$add', '$sub', '$mul', '$sdiv', '$smod', '$srem', '$udiv', '$urem', '$shl', '$lshr', '$ashr', '$and', '$or', '$xor', '$nand']
```

Comment:
Could we not have just one list containing names without `$` or `bv` prefixes. Then from that list you could add `$` or `bv` as needed. That would be better I think.